src/core/utils/otns.hpp

```diff
@@ -0,0 +1,138 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file wraps the calls to platform OTNS abstrations.
+ */
+
+#ifndef UTILS_OTNS_HPP_
+#define UTILS_OTNS_HPP_
+
+#include "openthread-core-config.h"
+
+#if (OPENTHREAD_MTD || OPENTHREAD_FTD) && OPENTHREAD_CONFIG_OTNS_ENABLE
+
+#include <openthread/thread.h>
+#include <openthread/thread_ftd.h>
+#include <openthread/platform/otns.h>
+
+#include "common/locator.hpp"
+#include "common/notifier.hpp"
+#include "mac/mac_types.hpp"
+#include "net/ip6_address.hpp"
+#include "thread/topology.hpp"
+
+namespace ot {
+namespace Utils {
+
+/**
+ * This class implements the OTNS Stub that interacts with OTNS.
+ *
+ */
+class OtnsStub : public InstanceLocator
```

Comment:
Can we consider removing the `Stub` (how about calling the class `Otns`?). The "stub" term is usually used as a [temporary/stand-in](https://en.wikipedia.org/wiki/Method_stub) (rather than main implementation).

Reply:
I named it to `OtnsStub` because this module serves as a client stub for interacting with `OTNS`, similar to a gRPC client stub that interacts with the gRPC server. Thoughts?